### PR TITLE
Rule 3.4 Block braces structure - correct errors

### PR DESCRIPTION
**Pull request by kentr0w**
### What's done:
  Made rule correct errors

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt [hunk 1]
```diff
@@ -123,27 +128,33 @@ class BlockStructureBraces : Rule("block-structure") {
...
     private fun checkOpenBraceOnSameLine(node: ASTNode, beforeType: IElementType, configuration: BlockStructureBracesConfiguration) {
         if (!configuration.openBrace) return
         val braceSpace = node.findChildByType(beforeType)?.treePrev
-        if (braceSpace == null || braceSpace.elementType != WHITE_SPACE || braceSpace.text.contains("\n".toRegex()))
+        if (braceSpace == null || checkBraceNode(braceSpace, true))
             BRACES_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "incorrect newline before opening brace",
                     (braceSpace ?: node).startOffset) {
-
+                if (braceSpace == null || braceSpace.elementType != WHITE_SPACE) {
+                    node.addChild(PsiWhiteSpaceImpl(" "), node.findChildByType(beforeType))
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt:149]: You have already called `findChildByType` once, you should either store result in a variable, or use `braceSpace.treeNext` here (according to how you created `braceSpace` variable.

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt [hunk 2]
```diff
@@ -168,6 +184,11 @@ class BlockStructureBraces : Rule("block-structure") {
             if (checkBraceNode(it, true)) {
                 BRACES_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "incorrect new line after closing brace",
                         it.startOffset) {
+                    if (it.elementType != WHITE_SPACE) {
+                        it.addChild(PsiWhiteSpaceImpl(" "), it.treePrev)
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt:188]: This line is marked as not covered by tests, please add an example for this case. Also, will this call work at all with `it.treePrev` as second argument?

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt [hunk 3]
```diff
@@ -123,27 +128,33 @@ class BlockStructureBraces : Rule("block-structure") {
...
+                }
             }
         checkOpenBraceEndLine(node, beforeType)
     }
 
-    private fun checkOpenBraceEndLine(node: ASTNode, beforeType: IElementType){
+    private fun checkOpenBraceEndLine(node: ASTNode, beforeType: IElementType) {
         val newNode = when (node.elementType) {
             IF -> {
                 when (beforeType) {
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt:160]: When you call `findChildByType(BLOCK)?.findChildByType(LBRACE)?.treeNext` I think it it safe to assume that `BLOCK` always has `LBRACE` and add `!!` there.

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt [hunk 4]
```diff
@@ -83,23 +89,28 @@ class BlockStructureBraces : Rule("block-structure") {
         val catchBlocks = tryBlock.catchClauses.map { it.node }
         val finallyBlock = tryBlock.finallyBlock?.node
         checkOpenBraceOnSameLine(tryBlock.node, BLOCK, configuration)
-        val allMiddleSpaceNode = node.findAllNodesWithSpecificType(FINALLY).map { it.treePrev } +
-                node.findAllNodesWithSpecificType(CATCH).map { it.treePrev }
-        checkMidBrace(allMiddleSpaceNode)
+        var allMiddleSpaceNode = node.findAllNodesWithSpecificType(CATCH).map { it.treePrev }
```
Comment (akuleshov7) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt:92]: allMiddleSpaceNode(s)

**Pull request merged**